### PR TITLE
New offers search

### PR DESCRIPTION
**Pull request by patrixr**
The offers search was previously running a local search on the store. And would fetch everything from the server through the `Find more on server` button, which was a huuuuuuuge payload.
Now the search is comparable to the order search on stock, just a normal endpoint call with the search text

### CODE REVIEW

File: server/mocks/offers.js [hunk 1]
```diff
@@ -1,112 +1,270 @@
...
-  var offers_json = {
-    "offers": [
-      { "id": "1", "state": "draft", "item_ids": ["1", "2"], "delivery_id": null, "created_by_id": "1" },
-      { "id": "2", "state": "draft", "delivery_id": null, "created_by_id": "1" },
-      { "id": "3", "language": "en", "state": "submitted",
-        "created_by_id": "1", "item_ids": ["4"], "delivery_id": null },
-      { "id": "4", "language": "en", "state": "under_review",
-        "created_by_id": "1", "item_ids": ["5"], "delivery_id": null,
-        "reviewed_by_id": "2" }
+  let emoty_search = {
```
Comment (abhinav098) [inline server/mocks/offers.js]: looks like a typo @patrixr .

File: server/mocks/offers.js [hunk 2]
```diff
@@ -1,112 +1,270 @@
...
+    res.statusCode = 404;
     res.send({});
   });
 
+  offersRouter.get("/search", function(req, res) {
+    let searchText = req.query["searchText"];
+    if (JSON.stringify(offers_json).indexOf(searchText) >= 0) {
+      return res.send(offers_json);
+    }
+    return res.send(emoty_search);
```
Comment (abhinav098) [inline server/mocks/offers.js]: here too

File: server/mocks/offers.js (hunk 1 above)
Comment (patrixr) [reply server/mocks/offers.js]: 😱

File: server/mocks/offers.js (hunk 2 above)
Comment (patrixr) [reply server/mocks/offers.js]: At least it was consistent 😆

**Comment by abulsayyad123** [comment]
👍

**Pull request merged**